modules/adding-ibm-z-lpar-agent-day-2.adoc

```diff
@@ -0,0 +1,124 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_with_agent_based_installer/prepare-pxe-assets-agent.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="adding-ibm-z-lpar-agents-day-2_{context}"]
+= Adding {ibm-z-title} agents in a Logical Partition (LPAR) as a day 2 operation
+
+In {product-title} {product-version}, the `.ins` and `initrd.img.addrsize` files are not automatically generated as part of the boot-artifacts for day 2. You must manually generate these files for {ibm-z-name} clusters running in an LPAR.
+
+The `.ins` file is a special file that includes installation data and is present on the FTP server. It can be accessed from the hardware management console (HMC) system. This file contains details such as mapping of the location of installation data on the disk or FTP server and the memory locations where the data is to be copied.
+
+.Prerequisites
+
+* A running file server with access to the Logical Partition (LPAR).
+
+.Procedure
+
+. To generate the `.ins` and `initrd.img.addrsize` files, follow these steps:
+
+.. Retrieve the size of the `kernel` and `initrd` by running the following commands:
++
+[source,terminal]
+----
+$ KERNEL_IMG_PATH='./kernel.img'
+----
++
+[source,terminal]
+----
+$ INITRD_IMG_PATH='./initrd.img'
+----
++
+[source,terminal]
+----
+$ CMDLINE_PATH='./generic.prm'
+----
++
+[source,terminal]
+----
+$ kernel_size=$(stat -c%s $KERNEL_IMG_PATH)
+----
++
+[source,terminal]
+----
+$ initrd_size=$(stat -c%s $INITRD_IMG_PATH)
+----
+
+.. Round up the `kernel` size to the next Mebibytes (MiB) boundary by running the following command:
++
+[source,terminal]
+----
+$ BYTE_PER_MIB=$(( 1024 * 1024 )) offset=$(( (kernel_size + BYTE_PER_MIB - 1) / BYTE_PER_MIB * BYTE_PER_MIB ))
+----
+
+.. Create the kernel binary patch file that contains the `initrd` address and size by running the following commands:
++
+[source,terminal]
+----
+$ INITRD_IMG_NAME=$(echo $INITRD_IMG_PATH | rev | cut -d '/' -f 1 | rev)
+----
++
+[source,terminal]
+----
+$ KERNEL_OFFSET=0x00000000
+----
++
+[source,terminal]
+----
+$ KERNEL_CMDLINE_OFFSET=0x00010480
+----
++
+[source,terminal]
+----
+$ INITRD_ADDR_SIZE_OFFSET=0x00010408
+----
++
+[source,terminal]
+----
+$ OFFSET_HEX=(printf '0x%08x\n' offset)
+----
+
+.. Convert the address and size to binary format by running the following command:
++
+[source,terminal]
+----
+$ printf "$(printf '%016x\n' $initrd_size)" | xxd -r -p > temp_size.bin
+----
+
+.. Merge the address and size binaries by running the following command:
+[source,terminal]
++
+----
+$ cat temp_address.bin temp_size.bin > "$INITRD_IMG_NAME.addrsize"
+----
+
+.. Clean up temporary files by running the following command:
+[source,terminal]
++
+----
+$ rm -rf temp_address.bin temp_size.bin
+----
+
+.. Create the `.ins` file by running the following command:
++
+[NOTE]
```

Comment:
Just checking - if this note isn't important to read before looking at the command block, can it be moved below it?